node/network/bridge/src/lib.rs

```diff
@@ -652,6 +719,23 @@ async fn run_network<N: Network>(
 					PeerSet::Collation => &mut collation_peers,
 				};
 
+				let maybe_authority = net.authority_discovery_service.get_authority_id_by_peer_id(peer.clone()).await;
+				if let Some(authority) = maybe_authority {
+					let mut ready = Vec::new();
+					for (i, pending) in pending_discovery_requests.iter_mut().enumerate() {
+						pending.on_authority_disconnected(&authority);
+						if pending.is_ready() {
+							ready.push(i);
+						}
+					}
+					// fulfill all ready requests and remove them from the pending list
+					for to_send in ready.into_iter().rev() {
+						pending_discovery_requests.swap_remove(to_send).send();
+					}
```

Comment:
This whole code chunk is duplicated above, maybe a candidate for a closure of `fn` :)

Reply:
the existence of duplicated code chunk is dependent on the second unresolved question: https://github.com/paritytech/polkadot/pull/1699#issuecomment-690966735, so I didn't want to introduce a premature abstraction :)